src/core/server/http/integration_tests/http_service.test.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import path from 'path';
+import request from 'request';
+import * as kbnTestServer from '../../../../test_utils/kbn_server';
+import { Router } from '../router';
+
+const authUrl = '/auth';
+const authHasSessionUrl = '/auth/has_session';
+const dummySecurityPlugin = path.resolve(__dirname, './__fixtures__/plugins/dummy_security');
+
+describe('http service', () => {
+  describe('setup contract', () => {
+    describe('#registerAuth()', () => {
+      let root: ReturnType<typeof kbnTestServer.createRoot>;
+      beforeAll(async () => {
+        root = kbnTestServer.createRoot(
+          {
+            plugins: { paths: [dummySecurityPlugin] },
+          },
+          {
+            dev: true,
+          }
+        );
+
+        const router = new Router('');
+        router.get({ path: authUrl, validate: false }, async (req, res) =>
+          res.ok({ content: 'ok' })
+        );
+        router.get({ path: authHasSessionUrl, validate: false }, async (req, res) =>
+          res.ok({ content: 'ok' })
+        );
+        // TODO fix me when registerRouter is available before HTTP server is run
+        (root as any).server.http.registerRouter(router);
```

Comment:
With the work going on in #18843, I think we'll be able to get the legacy code on the frontend more compatible with the lifecycle events in the new platform sooner than later. This would unblock adding back the `start` event.

Reply:
I believe it's fine to introduce `start` only on the server side for the time being. That can be implemented without leaking lifecycle semantic to plugins:
- the server runs `setup` for core services and plugins.
- the server runs `start` for core services only (until #18843 is ready).
- the server runs `legacy server`.